frontend/src/contexts/UserContext.tsx

```diff
@@ -0,0 +1,132 @@
+import { createContext, useContext } from 'react'
+import UserDTO from '../shared/Types/dtos/UserDTO';
+import { UnimplementedFunction } from '../utils';
+import { GoogleLoginResponse } from 'react-google-login';
+import AEYBResponse from '../shared/Types/responses/utils';
+import Permission from '../shared/Types/utils/Permission';
+import { useNavigate } from 'react-router-dom';
+import axios, { AxiosResponse } from 'axios';
+import useLocalStorage from '../hooks/UseLocastrorage';
+import API from '../shared/Types/api';
+
+// Eventually move to config file
+axios.defaults.baseURL = 'http://localhost:5000';
+
+export type FetcherType = <Url extends keyof API>(
+    url: Url, payload?: API[Url]["req"], params?: API[Url]["params"], queryParams?: API[Url]["query"]
+) => Promise<void | API[Url]["res"]>
+
+export interface UserContextProps {
+    user: UserDTO | null;
+    hasPermission: (permission: Permission | string) => boolean;
+    fetcher: FetcherType,
+    logout(): Promise<void>;
+    relogin(): Promise<void>;
+    onLogin(res: GoogleLoginResponse): Promise<void>;
+}
+
+export const useUserContext = () => useContext(UserContext);
+
+const UserContext = createContext<UserContextProps>({
+    user: null,
+    hasPermission: () => false,
+    fetcher: async () => UnimplementedFunction(),
+    logout: async () => UnimplementedFunction(),
+    relogin: async () => UnimplementedFunction(),
+    onLogin: async () => UnimplementedFunction(),
+})
+
+export function UserContextProvider({ children }) {
+    const navigate = useNavigate();
+
+    const [user, setUser] = useLocalStorage<UserDTO | null>('user', null);
+    const [permissions, setPermissions] = useLocalStorage<Permission[]>('permissions', []);
+    const [userToken, setUserToken] = useLocalStorage<string | null>('userToken', null)
+
+    async function relogin() {
+        if (userToken) {
+            axios.defaults.headers["Authorization"] = `Bearer ${userToken}`;
+            const data = await fetcher('GET /api/users/@me');
+            if (data) {
+                setUser(data.self);
+                setPermissions(data.permissions);
+                navigate('/', { replace: true })
+            }
+        }
+    }
+
+    async function onLogin(googleData: GoogleLoginResponse) {
+        const data = await fetcher('POST /api/users/login', { credential: googleData.tokenId });
+        if (data) {
+            axios.defaults.headers["Authorization"] = `Bearer ${data.token}`;
```

Comment:
Can you delete this line. The authorization is now set when you make the rest rather than globally to prevent errors when you refresh the page.